File: app/models/part_association.rb
# typed: false
# frozen_string_literal: true

class PartAssociation < ApplicationRecord

  attr_accessible :part_id, :collection_id, :row, :column

  belongs_to :part, class_name: 'Item'
  belongs_to :collection, class_name: 'Item'

  validate

end

File: legacy/app/assets/javascripts/models/part_association.js
AQ.PartAssociation.record_methods.upgrade = function(raw_data) {

  let pa = this;

  if ( raw_data.part ) {
    pa.part = AQ.Item.record(pa.part);
  }
  if ( raw_data.collection ) {
    pa.collection = AQ.Collection.record(pa.collection);
  }

}
